### ISSUE TITLE
`pod lib create` creates tests in example project, but not with pod. 

### ISSUE THREAD

**Issue by nikita-leonov** [CocoaPods/CocoaPods#3633]
Current implementation creates tests for pod library in example project, but not with library sources. Since tests outside of library module, it is impossible to test internal implementations.

Edit: Changed 'private implementations' to 'internal implementations'.

**Comment by segiddins** [comment]
Right, you should only be writing tests against your public interface.

**Comment by nikita-leonov** [comment]
My bad. Let me rephrase. Not private but internal and this request purely related to Swift.

**Comment by segiddins** [comment]
I understand. Swift's testing infrastructure right now is only mean for testing `public` things, and adding your Pod's source directly to the test target is a bad practice.

**Comment by nikita-leonov** [comment]
Are there plans to change this? I believe one of the reasons to have internal implementations is to be able to test everything within a module and expouse outside only an interface that end user will need.

**Comment by segiddins** [comment]
Anyways, this conversation belongs in https://github.com/CocoaPods/pod-template.
